Lib/test/datetimetester.py

```diff
@@ -2676,6 +2677,14 @@ def test_fromisoformat_fails_datetime(self):
                 with self.assertRaises(ValueError):
                     self.theclass.fromisoformat(bad_str)
 
+    def test_fromisoformat_fails_surrogate(self):
+        # Test that when fromisoformat() fails with a surrogate character as
+        # the separator, the error message contains the original string
+        dtstr = "2018-01-03\ud80001:0113"
+
+        with self.assertRaisesRegex(ValueError, re.escape(f"{dtstr!r}")):
```

Comment:
Perahps just `re.escape(repr(dtstr))`?

Reply:
Oops, thanks. This is a legacy from the fact that I wasn't sure if this had to match at the beginning of a line, so it was `f".*{dtstr!}"`.

Reply:
Fixed. Thanks.